src/foregroundTimeTracker.ts

```diff
@@ -0,0 +1,72 @@
+export default class ForegroundTimeTracker {
+    private isActive: boolean = false;
+    private localStorageName: string = '';
+    private startTime: number = 0;
+    private totalTime: number = parseFloat(
+        localStorage.getItem(this.localStorageName) || '0'
+    );
+
+    constructor(apiKey) {
+        this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
+        this.syncAcrossTabs = this.syncAcrossTabs.bind(this);
+        this.localStorageName = `mp-time-${apiKey}`;
+        this.init();
+    }
+
+    private init(): void {
+        document.addEventListener(
+            'visibilitychange',
+            this.handleVisibilityChange
+        );
+        window.addEventListener('beforeunload', () => this.updateTime());
+        window.addEventListener('storage', this.syncAcrossTabs); // Sync updates across tabs
+        this.startTracking();
+    }
+
+    private startTracking(): void {
+        if (!document.hidden) {
+            this.startTime = performance.now();
+            this.isActive = true;
+            setInterval(() => {
+                console.log(this.totalTime);
+            });
+        }
+    }
+
+    private stopTracking(): void {
+        if (this.isActive) {
+            this.updateTime();
+            this.isActive = false;
+        }
+    }
+
+    private updateTime(): void {
+        if (this.isActive) {
+            this.totalTime += (performance.now() - this.startTime) / 1000; // Convert ms to seconds
+            localStorage.setItem(
+                this.localStorageName,
+                this.totalTime.toFixed(2)
+            );
+            this.startTime = 0;
+        }
+    }
+
+    private handleVisibilityChange(): void {
+        if (document.hidden) {
```

Comment:
can we abstract the document out?  could this be some sort of "visibility handler"?